File: android/store/src/main/kotlin/com/cacompte/store/LeaderboardEntry.kt
package com.cacompte.store

import java.util.UUID

/**
 * Miroir de `LeaderboardEntry.swift` — un joueur dans le classement d'un jeu donné. Champs à
 * plat plutôt qu'une référence à [PlayerEntity], pour rester une valeur simple qui ne fait pas
 * fuiter un type Room hors de ce module.
 */
data class LeaderboardEntry(
    val playerID: UUID,
    val name: String,
    val avatarKind: String,
    val avatarValue: String,
    val avatarPhoto: ByteArray?,
    val paletteID: String,
    val played: Int,
    val wins: Int,
    val winRate: Double,
    /** `(nbJoueurs − rang) / (nbJoueurs − 1)`, moyenné — départage à taux de victoire égal. */
    val averageNormalizedRank: Double,
) {
    override fun equals(other: Any?): Boolean =
        other is LeaderboardEntry &&
            playerID == other.playerID &&
            name == other.name &&
            avatarKind == other.avatarKind &&
            avatarValue == other.avatarValue &&
            (avatarPhoto?.contentEquals(other.avatarPhoto) ?: (other.avatarPhoto == null)) &&
            paletteID == other.paletteID &&
            played == other.played &&
            wins == other.wins &&
            winRate == other.winRate &&
            averageNormalizedRank == other.averageNormalizedRank

    override fun hashCode(): Int = playerID.hashCode()
}

File: apple/QuiMeneKit/Sources/Store/LeaderboardEntry.swift
import Foundation

/// Doc 06 « Statistiques de groupe » (roadmap « après la v1 ») — un joueur dans le classement
/// d'un jeu donné. Champs à plat plutôt qu'une référence à `PlayerRecord` : même choix que
/// `ProfileStats.Nemesis`, pour rester `Sendable` sans faire fuiter un modèle SwiftData hors de
/// `Store`.
public struct LeaderboardEntry: Sendable, Equatable, Identifiable {
  public let playerID: UUID
  public let name: String
  public let avatarKind: String
  public let avatarValue: String
  public let avatarPhoto: Data?
  public let paletteID: String
  public let played: Int
  public let wins: Int
  public let winRate: Double
  /// `(nbJoueurs − rang) / (nbJoueurs − 1)`, moyenné — départage à taux de victoire égal,
  /// même mesure que `ProfileStats.averageNormalizedRank` (doc 06).
  public let averageNormalizedRank: Double

  public var id: UUID { playerID }

  public init(
    playerID: UUID,
    name: String,
    avatarKind: String,
    avatarValue: String,
    avatarPhoto: Data?,
    paletteID: String,
    played: Int,
    wins: Int,
    winRate: Double,
    averageNormalizedRank: Double
  ) {
    self.playerID = playerID
    self.name = name
    self.avatarKind = avatarKind
    self.avatarValue = avatarValue
    self.avatarPhoto = avatarPhoto
    self.paletteID = paletteID
    self.played = played
    self.wins = wins
    self.winRate = winRate
    self.averageNormalizedRank = averageNormalizedRank
  }
}
